### ISSUE TITLE
Doesn't build with the latest Swift Development Snapshot (DEVELOPMENT-SNAPSHOT-2020-02-21-a)

### ISSUE THREAD

**Issue by ikesyo** [Quick/Nimble#733]
https://travis-ci.org/github/Quick/Nimble/jobs/669964944

```
/home/travis/build/Quick/Nimble/Tests/NimbleTests/Matchers/PostNotificationTest.swift:17:9: error: ambiguous use of 'expect'
        expect {
        ^
/home/travis/build/Quick/Nimble/Sources/Nimble/DSL.swift:4:13: note: found this candidate
public func expect<T>(_ expression: @autoclosure @escaping () throws -> T?, file: FileString = #file, line: UInt = #line) -> Expectation<T> {
            ^
/home/travis/build/Quick/Nimble/Sources/Nimble/DSL.swift:13:13: note: found this candidate
public func expect<T>(_ file: FileString = #file, line: UInt = #line, expression: @escaping () throws -> T?) -> Expectation<T> {
            ^
```

Looks like this is regarding to two function overloads one with `@autoclosure` and the other with a (trailing) closure.

https://github.com/Quick/Nimble/blob/94ea5d1a4a4b095484ea2ca5d770a6252a5b5ee7/Sources/Nimble/DSL.swift#L3-L19

**Comment by ikesyo** [comment]
Confirmed that this is resolved now (tested in #738).

**Comment by ikesyo** [comment]
This was related to:

- https://bugs.swift.org/browse/SR-12291
- https://github.com/apple/swift-source-compat-suite/pull/418
